.eslintrc.js

```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  env: {
+    node: true,
+    es6: true
+  },
+  extends: [
+    'standard'
+  ],
+  parserOptions: {
+    ecmaVersion: 2018
```

Comment:
What's the difference between setting this and setting `es6: true`? If I understand correctly `es6: true` sets `ecmeVersion: 6`, but `2018` is equal to `ecmeVersion: 8` - so isn't there some sort of conflict here?

Reply:
If i remember correctly, this option is used by `espree` which is the default parser of eslint to understand the new syntax. While `env: 6` would allow using global variables without causing any lint errors. 

Some details here - https://eslint.org/docs/user-guide/configuring#specifying-parser-options

Reply:
Ok, as long as it's on purpose 👍 

I tried to look up what's needed to support Node.js 8 and nothing newer than that, but couldn't easily figure it out. Is this the right setup for Node.js 8?

Reply:
I guess yes `node`: true should pull in necessary globals. Unless we want to use new ones like `SharedArrayBuffer`, we may need to add them manually on globals property. 

Also, I am using eslint 7.0 which dropped support for Node 8 - should we need support it? https://eslint.org/blog/2020/05/eslint-v7.0.0-released#dropping-support-for-node-js-v8

Reply:
Using the latest eslint is fine with me - it doesn't need to run on Node.js 8 as long as it runs on one of our supported Node.js versions. The linting will be the same no matter which version it runs on.

What I think would be really cool would be if we could get it to lint for exact ECMAScript features we support. For example, we can't use the RegExp [`dotAll`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/dotAll) feature as that's not supported in all the versions of Node.js we support.

We need to target V8 6.0.287.53 which is the version that ships with Node.js v8.6.0. If I understand correctly, any version of V8 will not implement 100% of all ECMAScript editions, but say 100% of ES2015, 98% of ES2016, and so on. If eslint somehow was able to lint against this, that would be really nice so that we don't ship code that's not compatible with Node.js v8.6.0 by a mistake.

I'm not sure if `node: true` does this? After a bit of googling, I can see that there's an [eslint-plugin-node](https://www.npmjs.com/package/eslint-plugin-node) plugin which seems to use the `engines` field in `package.json` to lint for the minimum supported Node.js version?